xmodule/tasks.py

```diff
@@ -0,0 +1,319 @@
+            hashlib.sha1(str(source_key).encode('utf-8')).hexdigest()[:10]
+        )
+        return dest_parent_key.context_key.make_usage_key(source_key.block_type, block_id)
+
+    source_key = source_block.scope_ids.usage_id
+    new_block_key = generate_block_key(source_key, dest_parent_key)
+    try:
+        new_block = store.get_item(new_block_key)
+        if new_block.parent.block_id != dest_parent_key.block_id:
+            raise ValueError(
+                "Expected existing block {} to be a child of {} but instead it's a child of {}".format(
+                    new_block_key, dest_parent_key, new_block.parent,
+                )
+            )
+    except ItemNotFoundError:
+        new_block = store.create_child(
+            user_id,
+            dest_parent_key,
+            source_key.block_type,
+            block_id=new_block_key.block_id,
+        )
+
+    # Prepare a list of this block's static assets; any assets that are referenced as /static/{path} (the
+    # recommended way for referencing them) will stop working, and so we rewrite the url when importing.
+    # Copying assets not advised because modulestore doesn't namespace assets to each block like blockstore, which
+    # might cause conflicts when the same filename is used across imported blocks.
+    if isinstance(source_key, LibraryUsageLocatorV2):
+        all_assets = library_api.get_library_block_static_asset_files(source_key)
+    else:
+        all_assets = []
+
+    for field_name, field in source_block.fields.items():
+        if field.scope not in (Scope.settings, Scope.content):
+            continue  # Only copy authored field data
+        if field.is_set_on(source_block) or field.is_set_on(new_block):
+            field_value = getattr(source_block, field_name)
+            if isinstance(field_value, str):
+                # If string field (which may also be JSON/XML data), rewrite /static/... URLs to point to blockstore
+                for asset in all_assets:
+                    field_value = field_value.replace(f'/static/{asset.path}', asset.url)
+                    # Make sure the URL is one that will work from the user's browser when using the docker devstack
+                    field_value = blockstore_api.force_browser_url(field_value)
+            setattr(new_block, field_name, field_value)
+    new_block.save()
+    store.update_item(new_block, user_id)
+
+    if new_block.has_children:
+        # Delete existing children in the new block, which can be reimported again if they still exist in the
+        # source library
+        for existing_child_key in new_block.children:
+            store.delete_item(existing_child_key, user_id)
+        # Now import the children
+        for child in source_block.get_children():
+            _import_block(store, user_id, child, new_block_key)
+
+    return new_block_key
+
+
+def _filter_child(store, usage_key, capa_type):
+    """
+    Filters children by CAPA problem type, if configured
+    """
```

Comment:
```suggestion
def _filter_child(store, usage_key, capa_type):
    """
    Return whether this block is a problem of type `capa_type`.
    """
```

Docstring is wrong. Here's a more accurate one.